File: app/src/main/java/com/example/mytodolist/MainActivity.java
package com.example.mytodolist;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ListView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    Button btnadd;
    ListView lmain;
    ArrayList<String>list=new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btnadd=(Button) findViewById(R.id.btnadd);
        lmain=(ListView) findViewById(R.id.lmain);
        btnadd.setOnClickListener(this);

    }

    @Override
    public void onClick(View v) {
        Intent intent=new Intent(this,MainActivity2.class);
        startActivityForResult(intent, 1);
    }
    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (data == null) {return;}
        String text=data.getStringExtra("note");
        list.add(text);
        ArrayAdapter<String> adapter = new ArrayAdapter<String>(this,
                android.R.layout.simple_list_item_1, list);
        lmain.setAdapter(adapter);
    }
}

File: app/src/main/java/com/example/mytodolist/MainActivity.kt
package com.example.mytodolist

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.ListView
import androidx.activity.result.ActivityResultCallback
import androidx.activity.result.ActivityResultLauncher
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {
    var list = ArrayList<String>()
    val activityLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
        val data=result.data
        var text:String?=data?.getStringExtra("note")
        if (text != null) {
            list.add(text)
            val listNotes:ListView=findViewById(R.id.listNotes)
            val adapter=ArrayAdapter<String>(this,android.R.layout.simple_list_item_1,list)
            listNotes.adapter=adapter}
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val addButton: Button = findViewById(R.id.button2)
        addButton.setOnClickListener {
            val intent= Intent(this,MainActivity2::class.java)
            activityLauncher.launch(intent)}
    }

}
